### PR TITLE
Jigsaw-ify the CDK

### PR DESCRIPTION
**Pull request by egonw**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -504,10 +504,10 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-compiler-plugin</artifactId>
-                <version>2.3.2</version>
+                <version>3.7.0</version>
                 <configuration>
-                    <source>1.7</source>
-                    <target>1.7</target>
+                    <source>1.9</source>
```
Comment (tomas-pluskal) [inline pom.xml:509]: I believe for JDK versions 9+ this should only say `<source>9</source>`

**Comment by johnmay** [comment]
I would go to 11 actually..

**Comment by johnmay** [comment]
LTS versions are 8 and 11. Basically everyone is still using 8 - so be aware you are going to remove the CDK usability for a lot of people

**Comment by johnmay** [comment]
https://www.stackchief.com/blog/Which%20Version%20of%20Java%20Should%20You%20Use%3F

**Comment by tomas-pluskal** [comment]
> LTS versions are 8 and 11. Basically everyone is still using 8 - so be aware you are going to remove the CDK usability for a lot of people

If aiming for CDK 3 with the refactoring, then moving to Java 11 should not be an issue.

**Comment by egonw** [comment]
I'm trying to update the dependency to Java 11, but running into unexpected issue, like this one:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project cdk-interfaces: Fatal error compiling: release version 11 not supported -> [Help 1]
```

Cannot make sense of this, as the `maven-compile-plugin` 3.7.0 and higher is supposed to support Java 11.

**Comment by egonw** [comment]
Okay, nevermind. that was a matter of setting `JAVA_HOME`.

File: base/test-core/src/test/java/org/openscience/cdk/config/AtomTypeFactoryTest.java [hunk 2]
```diff
@@ -285,7 +285,7 @@ public void testGetAtomTypeFromMM2() throws Exception {
         Assert.assertEquals("S-[2];[H]{0,3}+=C.*+", atomType.getProperty(CDKConstants.SPHERICAL_MATCHER));
         Assert.assertEquals(Hybridization.SP2, atomType.getHybridization());
         Assert.assertTrue(atomType.getFlag(CDKConstants.IS_HYDROGENBOND_ACCEPTOR));
-        Assert.assertEquals(5, atomType.getProperty(CDKConstants.PART_OF_RING_OF_SIZE));
+        Assert.assertEquals(5, (int)atomType.getProperty(CDKConstants.PART_OF_RING_OF_SIZE));
```
Comment (johnmay) [inline base/test-core/src/test/java/org/openscience/cdk/config/AtomTypeFactoryTest.java:288]: What is the warning here? You should probably be doing this: 

```
Assert.assertEquals((Integer)5, atomType.<Integer>getProperty(CDKConstants.PART_OF_RING_OF_SIZE));
```
or 

```
Assert.assertThat(atomType.<Integer>getProperty(CDKConstants.PART_OF_RING_OF_SIZE),
                          CoreMatchers.is(5));
```

Comment (egonw) [reply base/test-core/src/test/java/org/openscience/cdk/config/AtomTypeFactoryTest.java:288]: I actually wanted the second, but it seems that requires an upgrade to JUnit5. The warning was that `assertEquals()` was ambiguous and the compiler did not know if it had to use `assertEquals(double, double)` or `assertEquals(Object, Object)`.

Comment (johnmay) [reply base/test-core/src/test/java/org/openscience/cdk/config/AtomTypeFactoryTest.java:288]: Second does not need Junit5

Comment (egonw) [reply base/test-core/src/test/java/org/openscience/cdk/config/AtomTypeFactoryTest.java:288]: It gave a warning that `Assert.assertThat()` was deprecated and wanted to focus first getting it compile. At this moment I'm looking into a NPE fail in `XYZWriterTest`.

Comment (johnmay) [reply base/test-core/src/test/java/org/openscience/cdk/config/AtomTypeFactoryTest.java:288]: Oh right yeah they changed it in the recent JUnit patch release. Change for the sake of change - sigh. It's now `Matcher.assertThat()` I think.

Comment (johnmay) [reply base/test-core/src/test/java/org/openscience/cdk/config/AtomTypeFactoryTest.java:288]: ```MatcherAssert.assertThat()``` even

**Comment by egonw** [comment]
Partially obsoleted by https://github.com/cdk/cdk/pull/652

**Comment by johnmay** [comment]
I would vote to close this, it's not clear what the changes are and making the compiler 1.9/9 is defo not correct - i.e. should be LTS 11 or 17. We may well one day be in a position to have packages in separate modules but this is just going to sit here and bit rot.

**Comment by johnmay** [comment]
Okay to leave the issue #626 open - I still stand by that the majority of the Java eco systems say "meh" to JDK modules :-)

**Comment by egonw** [comment]
For me the context was also further module clean up, but we just had a huge round of module clean up, so until we find new "issues" we close this one.

**Pull request closed without merging**